python/lsst/pipe/drivers/constructCalibs.py

```diff
@@ -834,10 +833,14 @@ def __init__(self, images):
                 self.background = np.nan
 
             def getCcdImage(self, detector, imageFactory, binSize):
-                detId = (detector.getId(),)
+                detId = detector.getId()
                 if detId not in self.images:
-                    return imageFactory(1, 1), detId
-                return self.images[detId], detId
+                    image = imageFactory(detector.getBBox())
+                    image.set(np.nan)
+                    image = afwMath.binImage(image, binSize)
```

Comment:
It's wasteful to create a full-size image of NaNs then bin it down.  Can't you create the binned image directly?